packages/plugin/src/plugin/data.ts

```diff
@@ -249,6 +262,37 @@ function matchesEntryPoint(
 	);
 }
 
+function modMatchesEntryPoint(
+	mod: JSONOutput.DeclarationReflection,
+	entry: PackageEntryConfig,
+	cfg: {
+		packagePath: string;
+		isSinglePackage: boolean;
+	},
+) {
+	const relModSources = mod.sources ?? [];
+	const relModSourceFile = relModSources.find((sf) => !!sf.fileName)?.fileName ?? '';
+	const relEntryPoint = joinUrl(cfg.packagePath, entry.path);
+	const isUsingDeepImports = !entry.path.match(/\.tsx?$/);
+
+	// Monorepos of 1 package don't have sources, so use the child sources
+	if (!relModSourceFile) {
+		const relSourceFiles = buildSourceFileNameMap(mod.children ?? []);
```

Comment:
wip: currently this will build a module map for each entryPoint, so don't love that. Probably needs to be lifted up. If so, I was thinking a lot of this logic could be skipped since all we care about is whether the entryPoint is part of this module (I think?).

Reply:
Maybe have it use `find()` so it short circuits earlier.

Reply:
I lifted it into a `Record<string, bool>` map to pass in so it keeps it a O(1) lookup, not that it makes a huge difference.